### PR TITLE
docs(components): [menu] fix invalid HTML code in the example

### PR DESCRIPTION
**Pull request by oliver139**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

`<div>` is not supposed to be the child of `<ul>`
This PR removed the gap created by `div.flex-grow` and instead using `margin-right: auto` to create the gap

### CODE REVIEW

**Comment by chenxch** [comment]
Thank you for your contribution, but this one does not seem to be as robust as the original demo, for example in rtl mode.

**Comment by oliver139** [comment]
> Thank you for your contribution, but this one does not seem to be as robust as the original demo, for example in rtl mode.

Oh yes, I really missed RTL. Will fix it.

**Comment by oliver139** [comment]
> Thank you for your contribution, but this one does not seem to be as robust as the original demo, for example in rtl mode.

But isn't that RTL mode is not well support in Element Plus yet?
At least the same example is currently also failed for having a wrong padding on the right to the logo on RTL mode.
The docs website is also not providing RTL language for now (and I believe that there is no plan for that?).

It would be hard and meaningless to handle RTL if the framework is not well supported or having no plan to support it (e.g. providing prop `:rtl="true"` for the components and adding `dir="rtl"` attribute on the components etc.).

File: docs/examples/menu/left-and-right.vue [hunk 1]
```diff
@@ -40,7 +39,7 @@ const handleSelect = (key: string, keyPath: string[]) => {
 </script>
 
 <style>
-.flex-grow {
-  flex-grow: 1;
+.el-menu--horizontal > .el-menu-item.img-logo {
```
Comment (btea) [inline docs/examples/menu/left-and-right.vue:42]: No need to add additional classes.
```suggestion
.el-menu--horizontal > .el-menu-item:nth-child(1) {
```

**Comment by tolking** [comment]
Why not just change the `div` to `li`?

**Comment by oliver139** [comment]
> Why not just change the `div` to `li`?

Then actually there will be an accessibility problem.

But if we talking about accessibility, I believe that the current HTML is also being a bit fail.
Where when a screen reader reading it, it would announce the number of item in the `<ul>` list.
If we include the logo into the list, it will announce there are 3 items in the list, where we should expect it counts the number of items in the menu only.

So actually I think the best approach would be:
```html
<div class="topbar">
  <div class="topbar__left">
    <!-- Logo -->
  </div>
  <div class="topbar__right">
    <!-- Menu -->
  </div>
</div>
```

```css
.topbar {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
}
```
But yes, it might already a bit too complicated for just a example

File: docs/examples/menu/left-and-right.vue (hunk 1 above)
Comment (oliver139) [reply docs/examples/menu/left-and-right.vue]: Yep, I did think about this too.

**Review by btea — APPROVED** [review]

**Review by tolking — APPROVED** [review]

**Comment by tolking** [comment]
Thank for your contribution

**Pull request merged**